core/shared/platform/linux-sgx/sgx_ipfs.c

```diff
@@ -291,8 +291,31 @@ ipfs_fopen(int fd, const char *filename, int flags)
         return NULL;
     }
 
-    // Opening the file
-    void *sgx_file = sgx_fopen_auto_key(filename, mode);
+    // Determine the symbolic link of the file descriptor, because IPFS does not
+    // support opening a relative path to a file descriptor (i.e., openat).
+    // Using the symbolic link in /proc/self allows to retrieve the same path as
+    // opened by the initial openat and respects the chroot of WAMR.
+    int size_of_path = snprintf(NULL, 0, "/proc/self/fd/%d", fd) + 1;
+    if (size_of_path < 0) {
+        errno = EINVAL;
+        return NULL;
+    }
+
+    char *symbolic_path = BH_MALLOC(size_of_path);
+    snprintf(symbolic_path, size_of_path, "/proc/self/fd/%d", fd);
```

Comment:
How about just defining a buf with enough size? Seems `char symbolic_path[32] = { 0 };` should be enough. Using BH_MALLOC/BH_FREE is relatively slow and not necessary?

Reply:
If BH_MALLOC/BH_FREE are slow, that totally makes sense to use a fixed-size array on the stack. I propose not to initialize it with zeroes, since `snprintf` takes care of adding the terminating null byte.

The new proposed code check for too-long symbolic paths and sets errno to `ENAMETOOLONG` if so.